integrations/jira/main.py

```diff
@@ -23,63 +35,59 @@ async def setup_application() -> None:
         )
         return
 
-    jira_client = JiraClient(
-        logic_settings["jira_host"],
-        logic_settings["atlassian_user_email"],
-        logic_settings["atlassian_user_token"],
-    )
-
-    await jira_client.create_events_webhook(
-        logic_settings["app_host"],
-    )
+    client = create_jira_client()
+    await client.create_events_webhook(app_host)
 
 
 @ocean.on_resync(ObjectKind.PROJECT)
 async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
-
+    client = create_jira_client()
     async for projects in client.get_paginated_projects():
-        logger.info(f"Received project batch with {len(projects)} issues")
+        logger.info(f"Received project batch with {len(projects)} projects")
         yield projects
 
 
 @ocean.on_resync(ObjectKind.ISSUE)
 async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
+    client = create_jira_client()
 
-    async for issues in client.get_paginated_issues():
+    config = cast(JiraResourceConfig, event.resource_config)
+    jql = config.selector.jql if config else None
+
+    async for issues in client.get_paginated_issues(jql):
         logger.info(f"Received issue batch with {len(issues)} issues")
         yield issues
 
 
+@ocean.on_resync(ObjectKind.TEAM)
+async def on_resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = create_jira_client()
+    org_id = ocean.integration_config.get("atlassian_organization_id")
+
+    if not org_id:
+        logger.info("Skipping team sync - no organization ID configured")
```

Comment:
I used logger.info here because org_id is an optional configuration parameter. If it's not provided, it's expected and valid that we skip syncing teams.
logger.info communicates what's happening (skipping team sync) and why (no org ID configured), without incorrectly implying that it's an error or warning situation.

What do you think?

Reply:
You make a brilliant point here. However, given that this is a situation that we want the user to detect, a warning log separates this log from the bunch of logs the integration generates and ensure it isn't lost in details.